From b7926c7bada91a333e2c49da83808589191b8a8f Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Fri, 7 May 2010 11:10:42 -0400
Subject: [PATCH] Explicitly add -lXext -lXau to linker line

Newer versions of GCC require all libraries to be explicitly
specified.

https://bugzilla.gnome.org/show_bug.cgi?id=583618
---
 configure.in              |    1 +
 gnome-session/Makefile.am |    2 ++
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/configure.in b/configure.in
index dcc2344..ef7c6dd 100644
--- a/configure.in
+++ b/configure.in
@@ -73,6 +73,7 @@ PKG_CHECK_MODULES(COMPAT,
 
 PKG_CHECK_MODULES(SM, sm)
 PKG_CHECK_MODULES(ICE, ice)
+PKG_CHECK_MODULES(XEXT, xext xau)
 
 PKG_CHECK_MODULES(DBUS_GLIB, dbus-glib-1 >= $DBUS_GLIB_REQUIRED)
 PKG_CHECK_MODULES(GCONF, gconf-2.0)
diff --git a/gnome-session/Makefile.am b/gnome-session/Makefile.am
index 5508f2e..120505c 100644
--- a/gnome-session/Makefile.am
+++ b/gnome-session/Makefile.am
@@ -14,6 +14,7 @@ INCLUDES =					\
 	$(DISABLE_DEPRECATED_CFLAGS)		\
 	$(SM_CFLAGS)				\
 	$(ICE_CFLAGS)				\
+	$(XEXT_CFLAGS)				\
 	$(GNOME_SESSION_CFLAGS)			\
 	$(GCONF_CFLAGS)				\
 	-I$(top_srcdir)/egg			\
@@ -51,6 +52,7 @@ gnome_session_LDADD =				\
 	$(GCONF_LIBS)				\
 	$(XRENDER_LIBS)				\
 	$(XTEST_LIBS)				\
+	$(XEXT_LIBS)				\
 	$(EXECINFO_LIBS)			\
 	$(NULL)
 
-- 
1.7.0.1

